src/Controller/CRUDController.php

```diff
@@ -1478,6 +1485,36 @@ final protected function checkParentChildAssociation(Request $request, object $o
         }
     }
 
+    private function getBatchActionExecutable(string $action): callable
+    {
+        $batchActions = $this->admin->getBatchActions();
+        if (!\array_key_exists($action, $batchActions)) {
+            throw new \RuntimeException(sprintf('The `%s` batch action is not defined', $action));
+        }
+
+        $controller = $batchActions[$action]['controller'] ?? sprintf(
+            '%s::%s',
+            $this->admin->getBaseControllerName(),
+            sprintf('batchAction%s', InflectorFactory::create()->build()->classify($action))
+        );
+
+        // This will throw an exception when called so we know if it's possible or not to call the controller.
+        $exists = false !== $this->container
+            ->get('controller_resolver')
+            ->getController(new Request([], [], ['_controller' => $controller]));
+
+        if (!$exists) {
+            throw new \RuntimeException(sprintf('Controller for action `%s` cannot be resolved', $action));
+        }
```

Comment:
@franmomu yes you are right but I am relying on the exception here for the test prior to execution of the query, the ask_confirmation, etc to tell the user the action is not configured properly. If we do it on a try and catch of the execution we would not easily know which exception to throw since it could be a exception from the execution. Once we move to version 5 we could just not catch the exception and have the default Symfony error stating the controller is not found and not bother about explaining how to configure the batch action controller.